test/support/warning_filter.rb

```diff
@@ -0,0 +1,53 @@
+require 'pathname'
+require 'set'
+
+module RegexpParserTest
+  # Stream filter for warnings
+  class WarningFilter
```

Comment:
This name was taken from mutant, and IMO it should be renamed to: `WarningExpectation` or something.